<a href="https://colab.research.google.com/github/Kavya0311/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Kavya0311/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

1. My Lane (or Freestyle) and Why:

I chose Lane 2: Refresh / Content Opportunity Scoring because it focuses on helping content and SEO teams identify which pages should be reviewed and refreshed first. Rather than manually checking thousands of pages, this project aims to prioritize pages based on observed performance data. I selected this lane because it addresses a practical business problem, supports data-driven decision-making, and provides an opportunity to build an interpretable machine learning solution that assists content maintenance.

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


2. Decision, Action, Cost of a Wrong Call

Decision:
Which content pages should be prioritized for refresh based on their observed performance?

Unit of Analysis:
One content page.

Output:

A ranked refresh opportunity score for each page, along with supporting reason codes and confidence.

Who Takes Action:
The SEO or content team reviews the ranked list and updates the highest-priority pages first.


Cost of a Wrong Recommendation:

If pages that truly need refreshing are missed, valuable search visibility and user engagement opportunities may be lost. If pages that do not require updates are prioritized, time and resources may be wasted.


Why Data and Machine Learning Help:

Content performance depends on multiple signals such as impressions, CTR, page age, and engagement. These relationships can be complex, making machine learning useful for combining many signals into a decision-support ranking that is more informative than simple manual rules.

In [6]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


3. Quick Look at the Data

To support my choice of Lane 2, I explored the starter dataset and calculated a few summary statistics. These observations help determine whether the dataset contains enough variation to support a content refresh prioritization problem.

In [7]:
!git clone https://github.com/Kavya0311/flyrank-ml-internship.git

fatal: destination path 'flyrank-ml-internship' already exists and is not an empty directory.


In [8]:
import pandas as pd

# Load the starter dataset
df = pd.read_csv("/content/flyrank-ml-internship/data/raw/content_refresh_anonymized.csv")

print("Total Pages:", len(df))

print("\nTrend Direction Counts:")
print(df["trend_direction"].value_counts())

print("\nAverage CTR:")
print(round(df["ctr"].mean(), 4))

print("\nAverage Impressions (90 Days):")
print(round(df["impressions_90d"].mean(), 2))



Total Pages: 30000

Trend Direction Counts:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Average CTR:
0.5107

Average Impressions (90 Days):
5200.37


Observation

The starter dataset contains 30,000 content pages. Among them, 16,262 pages are marked with a downward trend, indicating that a substantial number of pages may require attention. The dataset also has an average CTR of 0.5107 and an average of 5,200.37 impressions over the last 90 days. These observed statistics suggest that page performance varies considerably, making content refresh prioritization a meaningful decision-support problem for machine learning.

4. Careful Words: What I Can and Can't Claim:

This project provides decision support based on observed patterns in the available dataset. The results may indicate which pages are more likely to benefit from a content refresh, but they do not prove cause-and-effect relationships or predict Google's ranking algorithm. Any recommendations produced by this project should support human decision-making and be interpreted alongside domain expertise.

In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.